cs/src/core/Index/FASTER/FASTERImpl.cs

```diff
@@ -1842,6 +1836,40 @@ internal Status InternalContainsKeyInMemory(ref Key key, FasterExecutionContext
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void SynchronizeEpoch(FasterExecutionContext opCtx, FasterExecutionContext currentCtx, ref PendingContext pendingContext)
+        {
+            var version = opCtx.version;
+            Debug.Assert(currentCtx.version == version);
+            Debug.Assert(currentCtx.phase == Phase.PREPARE);
+            InternalRefresh(currentCtx);
+            Debug.Assert(currentCtx.version == version + 1);
+            Debug.Assert(currentCtx.phase == Phase.IN_PROGRESS);
+
+            pendingContext.version = currentCtx.version;
+        }
+
+        internal AsyncIOContext<Key, Value> ScheduleGetFromDisk(FasterExecutionContext opCtx,
+                    ref PendingContext pendingContext)
+        {
+            pendingContext.id = opCtx.serialNum;
```

Comment:
This results in a id of -1 in the beginning. Just fixing it to serialNum+1 is also not desirable because in future we will allow multiple ReadAsync operations to have the same serialNum. Therefore we have to use the `totalPending` field we had earlier to ensure ids are unique: `pendingContext.id = opCtx.totalPending++;`

Related, can you reconsider adding back the use of `HandleOperationStatus`? That will allow code to not diverge between the sync and async versions? If not, that is fine and we will take care of it after this PR. Right now, there is duplication of logic in two places which makes maintenance a nightmare.

Every line and variable in the original code is there for a reason, so please go through to see if any other issues like the above may have cropped up.

Reply:
> This results in a id of -1 in the beginning. Just fixing it to serialNum+1 is also not desirable because in future we will allow multiple ReadAsync operations to have the same serialNum. Therefore we have to use the totalPending field we had earlier to ensure ids are unique: pendingContext.id = opCtx.totalPending++;

Fixed by last commit.